lib/navigation.php

```diff
@@ -271,6 +271,11 @@ function gutenberg_migrate_menu_to_navigation_post( $new_name, $new_theme, $old_
 				'post_status'  => $post_status,
 			);
 			$navigation_post_id      = wp_insert_post( $post_data );
```

Comment:
We need to pass, true as the second param to wp_insert_post for it return a wp_error. 

Cc @adamziel

Reply:
Such a good catch, thank you @spacedmonkey ! I spun a PR in https://github.com/WordPress/gutenberg/pull/36483